src/placeos-frontend-loader/api/remote/github.cr

```diff
@@ -0,0 +1,137 @@
+require "hash_file"
+require "./remote"
+require "octokit"
+
+module PlaceOS::FrontendLoader
+  class Github < Remote
+    def initialize(@metadata : Metadata = Metadata.instance)
+    end
+
+    private alias Remote = PlaceOS::FrontendLoader::Remote
+
+    @github_client = Octokit.client(GIT_USER, GIT_PASS)
+
+    # Returns the branches for a given repo
+    def branches(repo : String) : Hash(String, String)
+      uri = "https://api.github.com/repos/#{repo}/branches"
+      response = HTTP::Client.get uri
+      raise Exception.new("status_code for #{uri} was #{response.status_code}") unless (response.success? || response.status_code == 302)
+      branches = Hash(String, String).new
+      Array(JSON::Any).from_json(response.body).map do |value|
+        branch_name = value["name"].to_s.strip.lchop("origin/")
+        branches[branch_name] = value["commit"]["sha"].to_s
+      end
+      branches
+    end
+
+    # Returns the commits for a given repo on specified branch
+    def commits(repo : String, branch : String) : Array(Remote::Commit)
+      url = "https://api.github.com/repos/#{repo}/commits?sha=#{branch}"
+      response = HTTP::Client.get url
+      raise Exception.new("status_code for #{url} was #{response.status_code}") unless (response.success? || response.status_code == 302)
+      commits = Array(Remote::Commit).new
+      Array(JSON::Any).from_json(response.body).map do |value|
+        commit = Remote::Commit.new(
+          commit: value["sha"].as_s,
+          name: value["commit"]["message"].as_s.strip(%(\n))
+        )
+        commits << commit
+      end
+      commits
+    end
+
+    def default_branch(repo : String) : String
+      url = "https://api.github.com/repos/#{repo}"
+      response = HTTP::Client.get url
+      raise Exception.new("status_code for #{url} was #{response.status_code}") unless (response.success? || response.status_code == 302)
+      parsed = JSON::Any.from_json(response.body)
+      parsed["default_branch"].to_s || "master"
```

Comment:
You're using strict indexing here, so the `|| "master"` won't do anything.

Reply:
```crystal
parsed["default_branch"]?.try(&.to_s) || "master"
```